libs/shared/providers/src/txButton/components/TxButton.tsx

```diff
@@ -38,10 +43,10 @@ export type TxButtonProps<
     functionName
   > = ContractFunctionArgs<abi, 'nonpayable' | 'payable', functionName>,
 > = {
-  label?: string;
+  label?: string | ((params: TxButtonFnParameters) => string | undefined);
```

Comment:
this is not what we discussed...

Reply:
I started down that other path and then realized by doing this I can easily handle the situation without having to hook into callbacks and hold state locally.